core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -1285,6 +1285,30 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
       }
     }
   }
+
+  test("SPARK-35691: addFile/addJar/addDirectory should put CanonicalFile") {
+    withTempDir { dir =>
+      try {
+        val sep = File.separator
+        val tmpDir = Utils.createTempDir(dir.getAbsolutePath + sep + "." + sep + ".")
```

Comment:
`Utils.createTempDir` seems return a canonical path. So, this test doesn't perform as you expect.
Also, it's better to assert the original path is not canonical.

Reply:
Yeah, update the test

Reply:
> Also, it's better to assert the original path is not canonical.

You add some assertions due to response to this comment right?
If so, I don't mean that.

I mean it's better to check whether `tmpDir` here is canonical or not.
In fact, what `Utils.createTempDir` returns is already canonical.

Reply:
is this correct?
```scala
 // Check those files are not canonical
assert(tmpAbsoluteDir.getAbsolutePath !== tmpAbsoluteDir.getCanonicalPath)
assert(tmpJar.getAbsolutePath !== tmpJar.getCanonicalPath)
assert(tmpFile.getAbsolutePath !== tmpFile.getCanonicalPath)
```

Reply:
Yeah.